File: campus-finder-backend/src/main/java/com/campus/finder/module/lost/entity/Claim.java
package com.campus.finder.module.lost.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 认领申请实体
 */
@Data
@TableName("biz_claim")
public class Claim implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 申请ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 失物招领记录ID
     */
    private Long lostFoundId;

    /**
     * 申请人ID
     */
    private Long userId;

    /**
     * 物品类型：found-捡到物品（申请认领），lost-丢失物品（提供线索）
     */
    private String claimType;

    // ========== 认领信息 ==========

    /**
     * 认领说明
     */
    private String description;

    /**
     * 物品特征描述（用于验证）
     */
    private String features;

    /**
     * 联系方式
     */
    private String contact;

    /**
     * 证明材料（图片）
     */
    private String proofImages;

    /**
     * 期望认领时间
     */
    private LocalDateTime expectTime;

    // ========== 审核状态 ==========

    /**
     * 申请状态：pending-待审核，approved-已通过，rejected-已拒绝，cancelled-已取消
     */
    private String status;

    /**
     * 审核备注（拒绝原因）
     */
    private String remark;

    /**
     * 审核人ID
     */
    private Long reviewedBy;

    /**
     * 审核时间
     */
    private LocalDateTime reviewedTime;

    // ========== 系统字段 ==========

    /**
     * 删除标记：0-未删除，1-已删除
     */
    @TableLogic
    private Integer deleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    // ========== 状态常量 ==========

    public static final String STATUS_PENDING = "pending";
    public static final String STATUS_APPROVED = "approved";
    public static final String STATUS_REJECTED = "rejected";
    public static final String STATUS_CANCELLED = "cancelled";

    // ========== 认领类型 ==========

    public static final String CLAIM_TYPE_CLAIM = "found";      // 认领捡到的物品
    public static final String CLAIM_TYPE_PROVIDE = "lost";    // 提供丢失物品线索
}

File: campus-finder-app/src/api/claim.js
import { get, post } from './request.js'

/**
 * 提交认领申请
 * @param {Object} data - { lostFoundId, claimType, description, features, contact, proofImages[] }
 */
export function submitClaimApi(data) {
  return post('/lost/claim/apply', data)
}

/**
 * 获取当前用户的认领申请列表
 */
export function getMyClaimsApi() {
  return get('/lost/claim/my')
}

/**
 * 获取认领申请详情
 * @param {number} id - 认领申请ID
 */
export function getClaimDetailApi(id) {
  return get(`/lost/claim/${id}`)
}

/**
 * 取消认领申请
 * @param {number} id - 认领申请ID
 */
export function cancelClaimApi(id) {
  return post(`/lost/claim/${id}/cancel`)
}

/**
 * 获取物品的所有认领申请（物品主人查看）
 * @param {number} lostFoundId - 物品ID
 */
export function getClaimsByItemApi(lostFoundId) {
  return get(`/lost/claim/lost-found/${lostFoundId}`)
}